confidential-data-hub/hub/src/hub.rs

```diff
@@ -6,19 +6,64 @@
 use std::collections::HashMap;
 
 use async_trait::async_trait;
+use image_rs::image::ImageClient;
 use kms::{Annotations, ProviderSettings};
 use log::info;
+use std::env;
 use storage::volume_type::Storage;
 
-use crate::{DataHub, Error, Result};
+use crate::{CdhConfig, DataHub, Error, Result};
 
 pub struct Hub {
     pub(crate) credentials: HashMap<String, String>,
+    image_client: tokio::sync::Mutex<image_rs::image::ImageClient>,
 }
 
 impl Hub {
-    pub async fn new(credentials: HashMap<String, String>) -> Result<Self> {
-        let mut hub = Self { credentials };
+    pub async fn new(config: CdhConfig) -> Result<Self> {
+        config.set_configuration_envs();
+        let credentials = config
+            .credentials
+            .iter()
+            .map(|it| (it.path.clone(), it.resource_uri.clone()))
+            .collect();
+
+        let mut image_client = ImageClient::new(config.image.work_dir.into());
+        image_client.config.auth = config.image.use_registry_credentials;
+        image_client.config.file_paths.auth_file = config.image.auth_uri;
+        image_client.config.file_paths.policy_path = config.image.policy_uri;
+        image_client.config.file_paths.sigstore_config = config.image.sigstore_config_uri;
+        image_client.config.max_concurrent_download = config.image.max_concurrent_download;
+        image_client.config.security_validate = config.image.security_validate;
+
+        if env::var("HTTPS_PROXY").is_err() {
+            let https_proxy = config.image.https_proxy;
+            if !https_proxy.is_empty() {
+                env::set_var("HTTPS_PROXY", https_proxy);
+            }
+        }
+
+        match env::var("HTTPS_PROXY") {
+            Ok(val) => info!("https_proxy is set to: {val}"),
+            Err(e) => log::warn!("failed to set https_proxy: {e}"),
+        };
```

Comment:
Yes I think you are right that we might probably only need only match statement. Btw, these messages would be printed by the CDH's stderr. In CoCo currently we do not have a way to expose, but in future we might need to redirect the logs of CDH and AA to somewhere for users.

In scenarios outside CoCo, it might be good to have these logs.